### ISSUE TITLE
Annotation array/element handling

### ISSUE THREAD

**Issue by bolismauro** [krzysztofzablocki/Sourcery#307]
I'm leveraging the fact that annotations can be used multiple times like this
```
// sourcery: variables=1, variables=2
```

The problem is that, if I use variable only once, the returned value in the stencil is an element and not a variable.

This forced to me to write really ugly code (and two different annotations)
```
    {% if case.annotations.variable %}
     // only a variable. Do something here

    {% else %}
      {% for v in case.annotations.variables %}
        // do something with v, which is part of an array

      {% endfor %}
    {% endif %}
```

Is there a better way to handle this? Like a check if it is an array, or transform a value into an array..

PS: Is there a release date for the new version of sourcery?

**Comment by bolismauro** [comment]
Most likely a duplicate of #302 but I'm not sure

**Comment by ilyapuchka** [comment]
Even if you iterate over annotations using key and value the value will be still either single value or array, so you will need to check it in the loop body. When you use just one annotation it does not make much sense to create array for it. So I would say having `if` in template is reasonable solution. To reuse code use [macro node](https://github.com/SwiftGen/StencilSwiftKit/blob/master/Documentation/tag-macro.md)

**Comment by bolismauro** [comment]
Can I pass a piece of code to the macro?
Because I have multiple things where I have that kind of check, but ideally I'd like to do something
```
{% call theHelper case.annotations.variables <a_block> %}
```

**Comment by ilyapuchka** [comment]
I think you can only pass objects in

**Comment by bolismauro** [comment]
Alright, so there isn't a good way to solve this problem. Got it :)

**Comment by ilyapuchka** [comment]
You can also use swift templates where you can define methods with any types of parameters.

**Comment by bolismauro** [comment]
Are they stable? The last time I checked they were not production ready (it was a long time ago)

 

On Sun, 21 May 2017 at 03:01 Ilya Puchka

<
mailto:Ilya Puchka <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

You can also use swift templates where you can define methods with any types of parameters.

—

You are receiving this because you authored the thread.

Reply to this email directly,
https://github.com/krzysztofzablocki/Sourcery/issues/307#issuecomment-302908201
, or

**Comment by ilyapuchka** [comment]
Thye were stable but slow, now the are stable and fast

**Comment by djbe** [comment]
@bolismauro Macro's are stored as object in the context stack, so you can pass them around just like any other objects.

**Comment by romdevios** [comment]
For string annotation variables it is possible to put extra empty element
```
// sourcery: variables="SomeName", variables=""
```
